### PR TITLE
Fixed issue with variable scope overriding suite result

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Fixed issue with variable scope overriding suite result

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [x] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: python-client/giskard/core/suite.py [hunk 1]
```diff
@@ -162,16 +163,7 @@ class TestPartial:
...
-    for r in test_results:
-        if type(r) == bool:
-            passed = passed and r
-        elif hasattr(r, "passed"):
-            passed = passed and r.passed
-        else:
-            logger.error(f"Invalid test result: {r.__class__.__name__}")
-            passed = False
-    return passed
+    return all(map(lambda res: res.passed, test_results))
```
Comment (mattbit) [inline python-client/giskard/core/suite.py:166]:
```py
not any(not res.passed for res in test_results)
```
will stop at the first failed test.

Comment (mattbit) [reply python-client/giskard/core/suite.py:166]: Let’s add a unit test for this. Can mock the rest result objects.

**Review by mattbit — CHANGES_REQUESTED** [review]

File: python-client/giskard/core/suite.py (hunk 1 above)
Comment (kevinmessiaen) [reply python-client/giskard/core/suite.py:166]: > ```python
> not any(not res.passed for res in test_results)
> ```
> 
> will stop at the first failed test.

Shouldn't `all` also stop at the first false? 
Something like this: `all(res.passed for res in test_results)`

**Review by mattbit — APPROVED** [review]

**Review by rabah-khalek — APPROVED** [review]
I've just verified it from the UI side, it works, thanks @kevinmessiaen

**Pull request merged**